pkg/provisioningv2/rke2/configserver/server.go

```diff
@@ -182,10 +179,16 @@ func (r *RKE2ConfigServer) connectConfigYaml(name, ns string, rw http.ResponseWr
 		return
 	}
 
+	kubernetesVersion, err := r.getClusterKubernetesVersion(mpSecret.Labels[capi.ClusterLabelName], ns)
+	if err != nil {
+		http.Error(rw, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
 	var content string
 	for _, f := range config["files"].([]interface{}) {
 		f := f.(map[string]interface{})
-		if path, ok := f["path"].(string); ok && path == fmt.Sprintf(planner.ConfigYamlFileName, "rke2") {
+		if path, ok := f["path"].(string); ok && path == fmt.Sprintf(planner.ConfigYamlFileName, rke2.GetRuntime(kubernetesVersion)) {
```

Comment:
I have a feeling the `rke2` was intentional as we do not support Windows with K3s and `configYaml` is for Windows to my knowledge.

@luthermonson may have more insight

Reply:
Two thoughts here:
1. Does it matter?
2. I think this will be going away in 2.6.4 because Windows nodes will use the planner.

Reply:
1. Nah, probably doesn't matter. 
2. Interesting, but makes sense